src/activation_cuda.cu

```diff
@@ -653,6 +653,84 @@ std::unique_ptr<BaseLayer> SoftmaxCuda::to_host()
     return host_layer;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// AGVI
+////////////////////////////////////////////////////////////////////////////////
+AGVICuda::AGVICuda() {}
```

Comment:
Any reason why we counldn't use this function to compute the `mu_a` and `var_a` so that we dont need to do it in the cuda kernel?

Reply:
I couldn't figure out why it was not working properly when doing it this way. Whenever we have the CPU version ready I will give it another try.

Reply:
ok, I will take a look when the CPU version is ready